### PR TITLE
#22 Fix is prop

### LINKED ISSUE #22: <Box /> Component does not support `is` propType
Hello,
Thanks for taking over this project.
When changing over to this, I noticed when I pass the `is` propType to the Box component, it no longer works. It just outputs the attribute without transforming the element underneath. Looking at the src/ I couldn't find any mention of the propType even though the Readme has it documented. Is this propType still supported?

Usage:
`<Box is="section">I'm a section tag</Box>`

Intended Result:
`<section>I'm a section tag</section>`

Actual Result:
`<div is="section">I'm a section tag</div>`

### PR DESCRIPTION
**Pull request by bringking**
### How does this PR make you feel (in animated GIF format)?
sweet

#### Prerequisites
- Is there any documentation that needs to be updated? N/A

#### Please provide a summary of changes and any background context...
This PR - 
* Fixes the `is` prop
* Rips out the test framework and replaces with `jest` and `jest-styled-components` to get readable output of the various CSS classes, which is very important for this tool

Closes #22 

---

### Technical Review
* [X] Is there enough test coverage?
* Ask questions, have a conversation!

#### How should this be manually tested (any data/API considerations)?
Checkout branch, run `npm run start` make sure docs work. Check the tests

### CODE REVIEW

File: src/propTypes.js [hunk 1]
```diff
@@ -3,6 +3,7 @@ import { oneOfType, number, string, array } from "prop-types";
 const responsivePropType = oneOfType([number, string, array]);
```
Comment (theshortcut) [inline src/propTypes.js]: should we downcase this file while we are at it?

Comment (bringking) [reply src/propTypes.js]: Yeah that would be good, will change that tomorrow

File: src/box.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    order: responsivePropType
+  };
+
+  static defaultProps = {
+    is: "div"
+  };
+
+  render() {
+    const {is, ...rest} = this.props;
+    const Base = styled(is)([], { boxSizing: "border-box" }, width, space, flex, order);
```
Comment (timurista) [inline src/box.js]: nice, now we can define the base component styled uses

File: src/box.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
+import React, { Component } from "react";
+import styled from "styled-components";
+import { space, width, responsiveStyle } from "styled-system";
+import { number, string, array, oneOfType } from "prop-types";
+import propTypes from "./propTypes";
+const responsivePropType = oneOfType([number, string, array]);
+const flex = responsiveStyle("flex");
+const order = responsiveStyle("order");
+
+class Box extends Component {
```
Comment (timurista) [inline src/box.js]: do we want flow typing here as a future improvement?

**Pull request merged**